src/main/java/com/google/research/bleth/simulator/Board.java

```diff
@@ -0,0 +1,58 @@
+package com.google.research.bleth.simulator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/** A container for all the agents, representing their locations, either real or estimated. */
+public class Board {
+    public final int rowNum;
+    public final int colNum;
+    ArrayList<Agent>[][] matrix;
+
+    public Board(int rows, int cols) {
+        rowNum = rows;
+        colNum = cols;
+        matrix = new ArrayList[rowNum][colNum];
+        for (int row = 0; row < rows; row++) {
+            Arrays.setAll(matrix[row], ArrayList::new);
+        }
+    }
+
+    /**
+     * Place an agent on board if the given location is valid.
+     * @param newLocation is the location where the agent will be placed.
+     * @param agent is the Agent which placed on the board.
+     */
+    public void placeAgent(Location newLocation, Agent agent) {
+        boolean isRowValid = newLocation != null && 0 <= newLocation.row && newLocation.row < rowNum;
```

Comment:
Consider using Guava Preconditions to simplify the error checking here.
https://guava.dev/releases/19.0/api/docs/com/google/common/base/Preconditions.html